server/mongodb/actions/Analytics.ts

```diff
@@ -0,0 +1,141 @@
+import Analytics from "@server/mongodb/models/Analytics";
+import { Days } from "@/common_utils/types";
+
+export const getAnalyticsByID = async (userID: string) => {
+  const analytics = await Analytics.findOne({ userID });
+  return analytics;
+};
+
+export const createAnalyticsID = async (userID: string) => {
+  const analytics = await Analytics.create({ userID });
+  return analytics;
+};
+
+// modify math function that sets and updates the object with the userID.
+export const modifyMath = async (
+  userID: string,
+  questionsAttempted: number,
+  questionsCorrect: number,
+  timePerQuestion: number,
+  difficultyScore: number,
+) => {
+  const weightedDiff =
+    (Number(questionsAttempted) / 100) * Number(difficultyScore);
```

Comment:
I implemented this just in case the user does one question on level 100 and 10 questions on level one. This is just to weight each difficulty score with the questions attempted

Reply:
Don't worry about this for the time being. We will store and use `difficultyScore` because we want to keep track of the change in `difficultyScore` across sessions and the weeks.